drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -372,16 +380,66 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 	}
 
 	if (i == clk_values) {
-		dev_err(bus->dev, "%s: could not find clock value for bandwidth %d\n",
+multilane:
+		dev_dbg(bus->dev, "%s: could not find clock value for bandwidth %d, checking multi-lane\n",
 			__func__, bus->params.bandwidth);
-		return -EINVAL;
+
+		m_rt = list_last_entry(&bus->m_rt_list, struct sdw_master_runtime, bus_node);
+		list_for_each_entry(s_rt, &m_rt->slave_rt_list, m_rt_node) {
+			slave_prop = &s_rt->slave->prop;
+
+			/* Find a non-zero manager lane */
+			for (i = 1; i < SDW_MAX_LANES; i++) {
+				if (!slave_prop->lane_maps[i])
```

Comment:
I think you should first check that all peripherals connected to that manager and using that stream do support the same lanes. Otherwise you may pick a lane that works for one of the peripherals but not for another.

This is a bit paranoid but it's not that crazy: we could have a big all-in-one device along with a companion amplifier on the same link. The big device will likely support multi-lane, the companion amplifier probably not.